src/cluster_asm.c

```diff
@@ -3393,28 +3405,10 @@ void asmActiveTrimCycle(void) {
     }
 }
 
-/* Trim a specific key if trimming is pending or in progress for its slot.
- * Return 1 if the key was trimmed */
-int asmActiveTrimDelIfNeeded(redisDb *db, robj *key, kvobj *kv) {
-    /* Check if trimming is in progress. */
-    if (server.allow_access_trimmed ||
-        !asmIsTrimInProgress())
-    {
-        return 0;
-    }
-
-    /* Check if the slot is in a trim job. */
-    sds keyname = key ? key->ptr : kvobjGetKey(kv);
-    if (!isSlotInTrimJob(getKeySlot(keyname)))
+/* Check if the key in a trim job. */
+int asmIsKeyInTrimJob(sds keyname) {
+    if (!asmIsTrimInProgress() || !isSlotInTrimJob(getKeySlot(keyname)))
         return 0;
-
-    if (key) {
-        asmActiveTrimDeleteKey(db, key);
-    } else {
-        robj *tmpkey = createStringObject(keyname, sdslen(keyname));
-        asmActiveTrimDeleteKey(db, tmpkey);
-        decrRefCount(tmpkey);
-    }
```

Comment:
we don't actually need to lazily trim the key, I don't remember why I thought this was a good idea.. perhaps just to copy existing behavior from the fork.

Reply:
> perhaps just to copy existing behavior from the fork.

guess so. before `REDISMODULE_CLUSTER_FLAG_NO_TRIM`, it is doesn't matter to trim the key here, right?

Reply:
I think it was harmless. I think the only path was calling `RM_OpenKey()` for a key in a slot that is being trimmed.